ndk-build/src/ndk.rs

```diff
@@ -196,22 +196,29 @@ impl Ndk {
         Ok((clang, clang_pp))
     }
 
-    pub fn toolchain_bin(&self, bin: &str, target: Target) -> Result<PathBuf, NdkError> {
+    pub fn toolchain_bin(&self, name: &str, target: Target) -> Result<PathBuf, NdkError> {
         #[cfg(target_os = "windows")]
         let ext = ".exe";
         #[cfg(not(target_os = "windows"))]
         let ext = "";
 
-        let bin = self.toolchain_dir()?.join("bin").join(format!(
-            "{}-{}{}",
-            target.ndk_triple(),
-            bin,
-            ext
-        ));
-        if !bin.exists() {
-            return Err(NdkError::PathNotFound(bin));
+        let toolchain_bin = self.toolchain_dir()?.join("bin");
+
+        // Since r21 (https://github.com/android/ndk/wiki/Changelog-r21) LLVM binutils are included _for testing_;
+        // Since r22 (https://github.com/android/ndk/wiki/Changelog-r22) GNU binutils are deprecated in favour of LLVM's;
+        // Since r23 (https://github.com/android/ndk/wiki/Changelog-r23) GNU binutils have been removed.
+        // To maintain stability with the current ndk-build crate release, prefer GNU binutils for
+        // as long as it is provided by the NDK instead of trying to use llvm-* from r21 onwards.
+        let gnu_bin = toolchain_bin.join(format!("{}-{}{}", target.ndk_triple(), name, ext));
+        if gnu_bin.exists() {
+            Ok(gnu_bin)
+        } else {
+            let llvm_bin = toolchain_bin.join(format!("llvm-{}{}", name, ext));
+            llvm_bin
+                .exists()
+                .then(|| llvm_bin)
+                .ok_or_else(|| NdkError::PathNotFound(gnu_bin))
```

Comment:
Not sure if we should print the GNU or LLVM bin in the error. Theoretically LLVM is only going to become relevant in the future, but for now we might want to have both... "Could not find <gnu binutils> tool nor <llvm binutils> tool".

Reply:
A dedicated error case would be nice imo

Reply:
I totally agree. How's this:

```console
me:android-ndk-rs/ $ cargo apk run -p ndk-examples
Error: GNU toolchain binary `aarch64-linux-android-ar` nor LLVM toolchain binary `llvm-ar` found in `"/home/marijn/Android/Sdk/ndk/23.0.7196353/toolchains/llvm/prebuilt/linux-x86_64/bin"`
```